### PR TITLE
Mark failing tests as xfail

### PR DESCRIPTION
**Pull request by sudo-panda**

### CODE REVIEW

**Comment by sudo-panda** [comment]
@vgvassilev please rerun the action after merging #9

**Comment by vgvassilev** [comment]
Can you rebase?

**Comment by vgvassilev** [comment]
We should also teach the CI to fail now if there are failing or xpassing tests.

**Comment by vgvassilev** [comment]
I meant we can rely on pytest’s exit code now. If anything unexpected happens the build should fail.

**Comment by sudo-panda** [comment]
It doesn't fail even when xpassed are present

**Comment by vgvassilev** [comment]
> It doesn't fail even when xpassed are present

I bet there is an option.

**Comment by vgvassilev** [comment]
> > It doesn't fail even when xpassed are present
> 
> I bet there is an option.

Here is some more information: https://docs.pytest.org/en/7.1.x/how-to/skipping.html#strict-parameter

And we could enable it by: `pytest --xfail_strict` which would make the default marker to strict causing the desired behavior.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -213,7 +213,7 @@ jobs:
         python -m pip install pytest
         python -m pip install pytest-xdist
         echo ::group::Test Logs
-        python -m pytest -n auto -sv --max-worker-restart 512 | tee test.log 2>&1
+        python -m pytest --xfail_strict -n auto -sv --max-worker-restart 512 | tee test.log 2>&1
```
Comment (vgvassilev) [inline .github/workflows/ci.yml:216]: To get the relevant exit code we should remove the pipe and the next lines or preserve the exit code.

Comment (sudo-panda) [reply .github/workflows/ci.yml:216]: The argument is not working

Comment (sudo-panda) [reply .github/workflows/ci.yml:216]: I can put strict in the mark itself

**Pull request merged**